client/state/sites/plans/actions.js

```diff
@@ -62,3 +63,21 @@ export function fetchSitePlansCompleted( siteId, plans ) {
 		plans: map( plans, createSitePlanObject )
 	};
 }
+
+/**
+ * Returns an action object to be used in updating an object containing
+ * the plans for the given site.
+ *
+ * @param  {Object} siteId ID of the concerned site
+ * @return {Object}        Action object
+ */
+export function refreshSitePlans( siteId ) {
+	return ( dispatch ) => {
+		dispatch( {
+			type: REMOVE_SITE_PLANS,
+			siteId
+		} );
+
+		return fetchSitePlans( siteId );
```

Comment:
In order for this function to fetch site plans, we need to pass the result of this call to `dispatch` like

```js
dispatch( fetchSitePlans( siteId ) );
```

This is because `refreshSitePlans` and `fetchSitePlans` are 'thunk action creators' ([docs](https://github.com/gaearon/redux-thunk)) and return functions that take `dispatch`.